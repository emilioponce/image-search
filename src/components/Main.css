.Main {
  margin-top: 20px;
  padding: 5px;
}
