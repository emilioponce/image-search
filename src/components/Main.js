import React, { Component } from 'react'

import Search from './Search'
import Gallery from './Gallery'

const Main = () => {
  return (
    <div>
      <Search />
      <Gallery />
    </div>
  )
}

export default Main
