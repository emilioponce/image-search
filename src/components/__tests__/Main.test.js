import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Main from '../Main'

// how to avoid JEST warnings -> https://github.com/facebook/jest/issues/5258
jest.mock('../../containers/Search', () => 'mock-search')
jest.mock('../../containers/Gallery', () => 'mock-gallery')
jest.mock('../../containers/ClearSearch', () => 'mock-clearSearch')
jest.mock('../../containers/Status', () => 'mock-status')

describe('Main component', () => {
  const component = <Main />

  it('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
