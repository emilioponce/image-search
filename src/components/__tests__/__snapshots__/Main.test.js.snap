// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component Snapshot matchs 1`] = `
<div
  className="Main"
>
  <div
    className="SearchArea"
  >
    <mock-search />
    <mock-clearSearch />
  </div>
  <mock-status />
  <mock-gallery />
</div>
`;
